diff --git a/windows/netuio/netuio.inf b/windows/netuio/netuio.inf
index 666a69c..64872de 100644
--- a/windows/netuio/netuio.inf
+++ b/windows/netuio/netuio.inf
@@ -28,6 +28,7 @@ HKR,,Icon,,-5
 
 [Standard.NT$ARCH$]
 %netuio.DeviceDesc%=netuio_Device, Root\netuio
+%netuio.nvme.DeviceDesc%=netuio_Device, PCI\CC_010802
 %Intel.F1572.Description%=netuio_Device, PCI\VEN_8086&DEV_1572
 %Intel.F1580.Description%=netuio_Device, PCI\VEN_8086&DEV_1580
 %Intel.F1581.Description%=netuio_Device, PCI\VEN_8086&DEV_1581
@@ -85,6 +86,7 @@ Provider = "Vendor"
 ClassName = "Windows UIO"
 DiskName = "DPDK netUIO Installation Disk"
 netuio.DeviceDesc = "netuio Device"
+netuio.nvme.DeviceDesc = "netuio NVM Express Controller"
 Intel.F1572.Description = "Intel(R) Ethernet Controller X710 for 10GbE SFP+"
 Intel.F1580.Description = "Intel(R) Ethernet Controller XL710 for 40GbE backplane"
 Intel.F1581.Description = "Intel(R) Ethernet Controller X710 for 10GbE backplane"
